### PR TITLE
Timeservice

### PR DESCRIPTION
**Pull request by roguishmountain**

### CODE REVIEW

File: test/unit/components/services/timeService.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                var time = TimeService.getTimer();
+                expect(time).to.be.a('object');
+                expect(time).to.have.property('time');
+                expect(time).to.have.property('day');
+                expect(time).to.have.property('displayDay');
+                expect(time).to.have.property('displayTime');
+            });
+        });
+
+        describe('getBasicTime', function() {
```
Comment (khipkin) [inline test/unit/components/services/timeService.js]: getBasicTimer

File: app/js/components/services/timeService.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    var getTimer = function() {
+        var timer = {
+            day: 0,
+            time: 0,
+            displayTime: "Time 00:00",
+            displayDay: "Day 0"
+        };
+        return timer;
+    };
+    var getBasicTimer = function() {
```
Comment (khipkin) [inline app/js/components/services/timeService.js]: this just repopulates every field of `getTimer()` with the same values that were already there

File: app/js/components/services/index.js [hunk 3]
```diff
@@ -3,4 +3,5 @@
 var app = require('angular').module('morojaApp');
 
 app.factory('StorageService', require('./storageService'));
-app.factory('MorojaConstantsService', require('./constantsService'));
\ No newline at end of file
+app.factory('MorojaConstantsService', require('./constantsService'));
+app.factory('TimeService', require('./timeService'));
```
Comment (khipkin) [inline app/js/components/services/index.js]: can you please add a nl to the end of this file

File: test/unit/components/services/timeService.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+describe('The TimeService\'s', function() {
+        var TimeService;
+
+        beforeEach(function() {
+            TimeService = new TimeServiceModule();
+        });
+
+        describe('getTimer', function() {
+            it('should return a time service', function() {
```
Comment (khipkin) [inline test/unit/components/services/timeService.js]: should return a time object

File: test/unit/village/services/villageService.js [hunk 5]
```diff
@@ -11,9 +11,11 @@ describe('The VillageService\'s', function() {
     var VillageService;
     var StorageService;
     var ConstantsService;
+    var TimeService;
     var village;
     var baseValues;
     var buildingStats;
+    var fakeTimer;
```
Comment (khipkin) [inline test/unit/village/services/villageService.js]: oops... i forgot to remove this line when i merged

**Comment by Arodang** [comment]
@roguishmountain @khipkin Is this ready to be merged in? Please do so if it is.

**Comment by khipkin** [comment]
I approve! That's what my :+1: was intended to convey. :smile:

File: test/unit/components/services/timeService.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+describe('The TimeService\'s', function() {
+        var TimeService;
+
+        beforeEach(function() {
+            TimeService = new TimeServiceModule();
+        });
+
+        describe('getBasicTimer', function() {
+            it('should return a basic time service with a time and day of zero', function() {
```
Comment (khipkin) [inline test/unit/components/services/timeService.js]: Just taking one last look, and you should add checks that the time and day are actually zero.

**Pull request merged**